agent-inject/agent/config_test.go

```diff
@@ -675,6 +675,17 @@ func TestConfigVaultAgentTemplateConfig(t *testing.T) {
 				MaxConnectionsPerHost: 100,
 			},
 		},
+		{
```

Comment:
Do we have a test for the default/unset condition?

Reply:
Updated with unset value